### PR TITLE
Add RMC generation fcl and test fcl, add photon conversion resampling

### PR DESCRIPTION
**Pull request by michaelmackenzie**
Adding generation and detector steps fcl for relevant RMC datasets and photon conversion resampling. This is in connection with [Offline PR 1371](https://github.com/Mu2e/Offline/pull/1371). The RMC generation can be tested with fcl added to `Tests/`, where for photon resampling a conversion point dataset must first be generated using `Tests/RMCGammaWriter.fcl`.

I'm sure I didn't follow the proper style in many places, so please let me know where the fcl should be better organized or if definitions should be moved to alternate places!

### CODE REVIEW

File: JobConfig/primary/IPAStopParticle.fcl
Comment (brownd1978) [inline JobConfig/primary/IPAStopParticle.fcl:11]: For future reference, to make a cut optional you can use art::OptionalAtom<> to encode both the value and whether the cut should be applied. That avoids having a separate bool and the possibility of a self-inconsistent config (value set but boolean not).

File: JobConfig/primary/RMC.fcl
Comment (brownd1978) [inline JobConfig/primary/RMC.fcl:2]: Do we need additional generator codes to distinguish RMC from the closure approximation vs tail generation to the kinematic limit?  ie between known and unknown physics?

File: JobConfig/primary/RMCExternal.fcl
Comment (brownd1978) [inline JobConfig/primary/RMCExternal.fcl:9]: empty line

File: JobConfig/primary/RMCGammaEndpointWriter.fcl
Comment (brownd1978) [inline JobConfig/primary/RMCGammaEndpointWriter.fcl:7]: These numbers also appear in RMCExternalEndpoint, can those scripts be nested to avoid the duplication?

File: JobConfig/primary/RMCInternalEndpoint.fcl
Comment (brownd1978) [inline JobConfig/primary/RMCInternalEndpoint.fcl:6]: more cases of repeated numbers and config.
If nesting doesn’t work could the value at least be set in prolog?

File: Tests/FlatGammaCaloSteps.fcl
Comment (brownd1978) [inline Tests/FlatGammaCaloSteps.fcl:2]: This run number isn’t inside the valid MDC2020 range and the results will not be processable (digitization and reconstruction) due to lack of conditions data. I’m not sure if that matters for this test.

File: Tests/RMCInternalSteps.fcl
Comment (brownd1978) [inline Tests/RMCInternalSteps.fcl:4]: I appreciate having test scripts, but there is a lot of replication. When the stops file evolves (which it will) this will require a lot of maintenance to update. Can the repeated parts be consolidated?

**Review by brownd1978 — COMMENTED** [review]
Overall this looks good, I point out a few targets for improvement.

File: JobConfig/primary/IPAStopParticle.fcl
Comment (michaelmackenzie) [reply JobConfig/primary/IPAStopParticle.fcl:11]: Interesting, I didn't realize this! I can make a quick PR to fix this, otherwise I'll make sure to do this next time.

File: JobConfig/primary/RMC.fcl
Comment (michaelmackenzie) [reply JobConfig/primary/RMC.fcl:2]: This is a fair question. I'm not sure what's the best choice here, it seems reasonable to have the tail measurement connected to its own process code so I'm happy to implement one if recommended.

File: JobConfig/primary/RMCExternal.fcl
Comment (michaelmackenzie) [reply JobConfig/primary/RMCExternal.fcl]: I added a line here.

File: JobConfig/primary/RMCGammaEndpointWriter.fcl
Comment (michaelmackenzie) [reply JobConfig/primary/RMCGammaEndpointWriter.fcl]: I moved the measured and kinematic limit endpoints to the prolog and then reference them here, so there's now only one maintenance point if we want to update the values. Does this seem like a better setup?

File: JobConfig/primary/RMCInternalEndpoint.fcl
Comment (michaelmackenzie) [reply JobConfig/primary/RMCInternalEndpoint.fcl]: I moved this to the prolog.

File: Tests/FlatGammaCaloSteps.fcl
Comment (michaelmackenzie) [reply Tests/FlatGammaCaloSteps.fcl]: Is there a better run number to use for testing? I took this from an existing test script.

File: Tests/RMCInternalSteps.fcl
Comment (michaelmackenzie) [reply Tests/RMCInternalSteps.fcl]: I moved the muon stop file info (and the run number) to a test fcl file `Production/Tests/MuonStopConfig.fcl` so the muon stop file can more easily be updated. I also updated the CE, DIO, and flat e+/e- test fcl to include this config file to reduce maintenance.

File: Tests/FlatGammaCaloSteps.fcl
Comment (sophiemiddleton) [reply Tests/FlatGammaCaloSteps.fcl]: you need something 1200-1210

File: JobConfig/primary/IPAStopParticle.fcl
Comment (brownd1978) [reply JobConfig/primary/IPAStopParticle.fcl:11]: If you can patch this quickly in Offline that would be best.

File: JobConfig/primary/RMC.fcl
Comment (brownd1978) [reply JobConfig/primary/RMC.fcl:2]: Lets defer this to when you have a 'RMC tail' generator.

File: Tests/FlatGammaCaloSteps.fcl
Comment (brownd1978) [reply Tests/FlatGammaCaloSteps.fcl]: But don't use 1205, that's for the extracted position.

**Review by brownd1978 — APPROVED** [review]
Thanks for the updates. I'm approving as complete, if you want to update to use OptionalAtom now that's fine. I'd like at least 1 other person (Sophie, Yuri) to review before merging.

File: Tests/FlatGammaCaloSteps.fcl
Comment (michaelmackenzie) [reply Tests/FlatGammaCaloSteps.fcl]: I updated the test fcl to use run 1210 instead

**Pull request merged**
